Gems/ToolingConnection/Code/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+
+# Currently we are in the Code folder: ${CMAKE_CURRENT_LIST_DIR}
+# Get the platform specific folder ${pal_dir} for the current folder: ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME}
+# Note: o3de_pal_dir will take care of the details for us, as this may be a restricted platform
+#       in which case it will see if that platform is present here or in the restricted folder.
+#       i.e. It could here in our gem : Gems/ToolingConnection/Code/Platform/<platorm_name>  or
+#            <restricted_folder>/<platform_name>/Gems/ToolingConnection/Code
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_name}")
+
+# Now that we have the platform abstraction layer (PAL) folder for this folder, thats where we will find the
+# traits for this platform. Traits for a platform are defines for things like whether or not something in this gem
+# is supported by this platform.
+include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
+
+# Add the ToolingConnection.Static target
+# Note: We include the common files and the platform specific files which are set in toolingconnection_common_files.cmake
+# and in ${pal_dir}/toolingconnection_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+ly_add_target(
+    NAME ToolingConnection.Static STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        toolingconnection_files.cmake
+        toolingconnection_autogen_files.cmake
+        ${pal_dir}/toolingconnection_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            AZ::AzNetworking
+    AUTOGEN_RULES
+        *.AutoPackets.xml,AutoPackets_Header.jinja,$path/$fileprefix.AutoPackets.h
+        *.AutoPackets.xml,AutoPackets_Inline.jinja,$path/$fileprefix.AutoPackets.inl
+        *.AutoPackets.xml,AutoPackets_Source.jinja,$path/$fileprefix.AutoPackets.cpp
+        *.AutoPackets.xml,AutoPacketDispatcher_Header.jinja,$path/$fileprefix.AutoPacketDispatcher.h
+        *.AutoPackets.xml,AutoPacketDispatcher_Inline.jinja,$path/$fileprefix.AutoPacketDispatcher.inl
+)
```

Comment:
This is missing the ToolingConnection.API target that comes with the Default Gem Template.

If another gem is to depend on this gem API it would depend on the ToolingConnection.API target

Reply:
Ideally most of the interaction will be driven through the interface in AzFramework this Gem implements so I'm not entirely sure it will need it yet.

Reply:
In that case can you rename the `ToolingConnection.Static` target to `ToolingConnection.Private.Static`.

We don't want outside gem users to depend on the static target then.